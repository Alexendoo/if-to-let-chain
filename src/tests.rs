use insta::{Settings, assert_snapshot};
use std::fs;

macro_rules! snaps {
    ($($file:ident)*) => {
        $(
            #[test]
            fn $file() {
                let path = concat!("src/inputs/", stringify!($file), ".rs");
                let mut settings = Settings::new();
                settings.set_prepend_module_to_snapshot(false);
                settings.set_input_file(path);
                settings.bind(|| {
                    let mut contents = fs::read_to_string(path).unwrap();
                    super::modify(&mut contents, 4, path);
                    assert_snapshot!(contents);
                });
            }
        )*
    };
}

snaps!(
    closure
    comment
    or
    simple
);
