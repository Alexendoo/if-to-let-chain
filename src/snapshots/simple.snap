---
source: src/tests.rs
expression: contents
input_file: src/inputs/simple.rs
---
fn main() {
    if x == 1
        && let Some(y) = f()
    {
        stuff
    } else {
        other + stuff
    }
}
